sdk/core/core-lro/test/utils/router/paramRoutes.ts

```diff
@@ -4,380 +4,402 @@
-      if (deleteasyncRetry.internalCounter === 0) {
-        return buildResponse(request, finalCode, `{ "status": "${getPascalCase(finalState)}"}`);
-      } else {
-        --deleteasyncRetry.internalCounter;
-        const pollingUri = `/deleteasync/${retry}/${finalState.toLowerCase()}/operationResults/${finalCode}`;
-        const headers = createHttpHeaders({
-          "Azure-AsyncOperation": pollingUri,
-          Location: pollingUri,
-        });
-        if (retry === "retry") {
-          headers.set("Retry-After", "0");
+      if (pieces.length === 6 && pieces[0] === "delete" && pieces[1] === "provisioning") {
+        const initialCode = JSON.parse(pieces[2]);
+        const initialState = getPascalCase(pieces[3]);
+        const finalCode = JSON.parse(pieces[4]);
+        const finalState = getPascalCase(pieces[5]);
+        if (!isValidRequest(initialCode, initialState, finalState, finalCode)) return undefined;
+        if (request.method === "DELETE") {
+          return {
+            request: request,
+            headers: createHttpHeaders({
+              Location: `/delete/provisioning/${initialCode}/${initialState.toLowerCase()}/${finalCode}/${finalState.toLowerCase()}`,
+              "Retry-After": "0",
+            }),
+            status: initialCode,
+            bodyAsText: `{ "properties": { "provisioningState": "${initialState}"}, "id": "100", "name": "foo" }`,
+          };
+        } else if (request.method === "GET") {
+          return buildResponse(
+            request,
+            finalCode,
+            `{ "properties": { "provisioningState": "${finalState}"}, "id": "100", "name": "foo" }`
+          );
         }
-        return buildResponse(request, 202, `{ "status": "Accepted"}`, headers);
       }
     } catch (e) {
       return undefined;
     }
+    return undefined;
   }
-  return undefined;
-}
-
-// eslint-disable-next-line @typescript-eslint/no-namespace
-namespace deleteasyncRetry {
-  export let internalCounter: number = 1; // eslint-disable-line prefer-const
 }
 
-function putasyncRetry(request: PipelineRequest): PipelineResponse | undefined {
-  function isValidRequest(retry: string, finalState: string): boolean {
-    return (
-      (retry === "retry" || retry === "noretry") &&
-      (finalState === "succeeded" || finalState === "canceled" || finalState === "failed")
-    );
-  }
-  const pieces = parseUri(request.url);
-  if (pieces[0] !== "putasync") {
-    return undefined;
-  }
-  const retry = pieces[1];
-  const finalState = pieces[2];
-  if (!isValidRequest(retry, finalState)) return undefined;
-  if (pieces.length === 3) {
-    const method = request.method;
-    if (method === "GET") {
-      return buildResponse(
-        request,
-        200,
-        `{ "properties": { "provisioningState": "${getPascalCase(
-          finalState
-        )}"}, "id": "100", "name": "foo" }`
+function createDeleteAsyncRetry(): Processor {
+  let internalCounter = 1;
```

Comment:
I am still not clear what's the benefit of a list of objects over the old list of functions? the state is local to the function, right?

Reply:
We talked about it offline and agreed on the current approach.